pkg/corerp/backend/deployment/deploymentprocessor_test.go

```diff
@@ -301,21 +301,20 @@ func Test_Render(t *testing.T) {
 		},
 	}
 
-	mocks := setup(t)
-	dp := deploymentProcessor{mocks.model, mocks.dbProvider, nil, nil}
-
 	t.Run("verify render success", func(t *testing.T) {
+		mocks := setup(t)
+		dp := deploymentProcessor{mocks.model, mocks.dbProvider, nil, nil}
+
 		testResource := getTestResource()
 		testRendererOutput := getTestRendererOutput()
 		resourceID := getTestResourceID(testResource.ID)
 
-		depId1, _ := resources.ParseResource("/subscriptions/test-subscription/resourceGroups/test-resource-group/providers/Applications.Core/httpRoutes/A")
-		depId2, _ := resources.ParseResource("/subscriptions/test-subscription/resourceGroups/test-resource-group/providers/Applications.Datastores/mongoDatabases/test-mongo")
-		requiredResources := []resources.ID{depId1, depId2}
+		depId1, _ := resources.ParseResource("/subscriptions/test-subscription/resourceGroups/test-resource-group/providers/Applications.Datastores/mongoDatabases/test-mongo")
+		requiredResources := []resources.ID{depId1}
 
 		mocks.renderer.EXPECT().Render(gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(testRendererOutput, nil)
 		mocks.renderer.EXPECT().GetDependencyIDs(gomock.Any(), gomock.Any()).Times(1).Return(requiredResources, nil, nil)
-		mocks.dbProvider.EXPECT().GetStorageClient(gomock.Any(), gomock.Any()).Times(5).Return(mocks.db, nil)
+		mocks.dbProvider.EXPECT().GetStorageClient(gomock.Any(), gomock.Any()).AnyTimes().Return(mocks.db, nil)
```

Comment:
Should we specify the exact number of times that this is going to be called?

Reply:
eg: why is 4 the right number? would we have a bug if the value was 8? or 3?

Reply:
I think the reason why 4 or 5 is right is because that is the exact number of times that this function should be called for the scenario that is being tested. If we don't specify a number, then we won't be able to catch the case if this function is being called 6 times instead of 5. Because if it is being called more or less than expected then there should be a problem with our test.